gen2-human-machine-safety/main.py

```diff
@@ -39,31 +44,26 @@ def __init__(self):
         self.depthWidth = 1080.0
 
     # Calculate spatial coordinates from depth map and bounding box (ROI)
-    def calc_spatials(self, bbox, depth):
+    def calc_spatials(self, bbox, depth, averaging_method=np.mean):
+        xmin, ymin, xmax, ymax = bbox
         # Decrese the ROI to 1/3 of the original ROI
-        deltaX = int((bbox[2] - bbox[0]) * 0.33)
-        deltaY = int((bbox[3] - bbox[1]) * 0.33)
-        bbox[0] = bbox[0] + deltaX
-        bbox[1] = bbox[1] + deltaY
-        bbox[2] = bbox[2] - deltaX
-        bbox[3] = bbox[3] - deltaY
-
-        # Calculate the average depth in the ROI. TODO: median, median /w bins, mode
-        cnt = 0.0
-        sum = 0.0
-        for x in range(bbox[2] - bbox[0]):
-            for y in range(bbox[3] - bbox[1]):
-                depthPixel = depth[bbox[1] + y][bbox[0] + x]
-                if DEPTH_THRESH_LOW < depthPixel and depthPixel < DEPTH_THRESH_HIGH:
-                    cnt+=1.0
-                    sum+=depthPixel
-
-        averageDepth = sum / cnt if 0 < cnt else 0
+        deltaX = int((xmax - xmin) * 0.33)
+        deltaY = int((ymax - ymin) * 0.33)
+        xmin += deltaX
+        ymin += deltaY
+        xmax -= deltaX
+        ymax -= deltaY
+
+        # Calculate the average depth in the ROI.
+        depthROI = depth[ymin:ymax, xmin:xmax]
```

Comment:
Sometimes the detection bbox is flipped (e.g. `bbox` variable can be `[720, 720, 0, 0]`), so cutting the frame before check throws an error at random during runtime